src/CommunityToolkit.Maui/Views/Popup/PopupExtensions.shared.cs

```diff
@@ -63,6 +63,7 @@ void handler(object? sender, EventArgs args)
 
 		async void handler(object? sender, EventArgs args)
 		{
+			page.GetCurrentPage().Unloaded -= (_, _) => { };
```

Comment:
this will not work as expected, the compiler will create a new delegate to represent this one and it will be different from the delegate subscribed at line `81`. In order words, it will never unsubscribe the various `Unloaded` registrations causing a memory leak.
The fix is to create a local variable that will represent this delegate and use it

Reply:
Yep I was in doubt about that. Will fix it.